File: Ejercicios Propuestos 3/Ejercicio 21/Ejercicio 21.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

def ordenar_pares_impares(arr):
    n = len(arr)

    # Aplicar el método Bubble Sort
    for i in range(n - 1):
        for j in range(n - 1 - i):

            debe_intercambiar = False

            # Si el izquierdo es impar y el derecho es par
            if arr[j] % 2 != 0 and arr[j + 1] % 2 == 0:
                debe_intercambiar = True

            # Si ambos son pares, ordenar ascendentemente
            elif arr[j] % 2 == 0 and arr[j + 1] % 2 == 0:
                if arr[j] > arr[j + 1]:
                    debe_intercambiar = True

            # Si ambos son impares, ordenar descendentemente
            elif arr[j] % 2 != 0 and arr[j + 1] % 2 != 0:
                if arr[j] < arr[j + 1]:
                    debe_intercambiar = True

            # Realizar el intercambio
            if debe_intercambiar:
                aux = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = aux


# Programa principal

# Solicitar la cantidad de elementos
n = int(input("Ingrese la cantidad de elementos que tendrá el vector: "))

# Crear el vector
vector = []

# Llenar el vector
print("--- Ingrese los números del vector ---")

for i in range(n):
    numero = int(input(f"Valor para la posición {i}: "))
    vector.append(numero)

# Mostrar el vector original
print("\nVector original:")

for numero in vector:
    print(numero, end=" ")

print("\n----------------------")

# Ordenar el vector
ordenar_pares_impares(vector)

# Mostrar el resultado
print("Vector final ordenado:")

for numero in vector:
    print(numero, end=" ")

print()

File: Ejercicios Propuestos 3/Ejercicio 21/ejercicio21.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Ingrese la cantidad de elementos: ";
    cin >> n;

    int v[100];
    for (int i = 0; i < n; i++) {
        cout << "Elemento " << i << ": ";
        cin >> v[i];
    }

    int pares[100], impares[100];
    int cp = 0, ci = 0;

    for (int i = 0; i < n; i++) {
        if (v[i] % 2 == 0) {
            pares[cp++] = v[i];
        } else {
            impares[ci++] = v[i];
        }
    }

    // Ordenar pares ascendente
    for (int i = 0; i < cp - 1; i++) {
        for (int j = 0; j < cp - 1 - i; j++) {
            if (pares[j] > pares[j + 1]) {
                int temp = pares[j];
                pares[j] = pares[j + 1];
                pares[j + 1] = temp;
            }
        }
    }

    // Ordenar impares descendente
    for (int i = 0; i < ci - 1; i++) {
        for (int j = 0; j < ci - 1 - i; j++) {
            if (impares[j] < impares[j + 1]) {
                int temp = impares[j];
                impares[j] = impares[j + 1];
                impares[j + 1] = temp;
            }
        }
    }

    cout << "Resultado: ";
    for (int i = 0; i < cp; i++) cout << pares[i] << " ";
    for (int i = 0; i < ci; i++) cout << impares[i] << " ";
    cout << endl;

    return 0;
}
